pyout/field.py

```diff
@@ -138,6 +201,17 @@ def truncate_fn(_, result):
             return result[:length]
         return truncate_fn
 
+    @staticmethod
+    def transform(function):
+        """Return a processor for a style's "transform" function.
+        """
+        def transform_fn(_, result):
+            try:
+                return function(result)
+            except:
+                raise StyleFunctionError(function)
```

Comment:
It doesn't hide it.  It's above (and `StyleFunctionError`'s message
even says to look above).  Or perhaps this is python version
dependent?

Reply:
"above" what? (if you mean a backtrace -- that is not what I mean, I mean an actual exception -- what went wrong exactly?)
 the original exception is caught and not anyhow printed if I got it right.  Actually probably no need to get a dedicated catching and you could use `sys.exc_info()` within StyleFunctionError constructor to collect/present the most recent exception

Reply:
> if you mean a backtrace -- that is not what I mean, I mean an actual exception -- what went wrong exactly?

yes, I mean the traceback which includes the exception.

For example, running

```python
from pyout import Tabular
from humanize import naturalsize

out = Tabular(columns=["name", "size", "path"],
              style={
                  "separator_": " | ",
                  "size": {"transform": naturalsize}})

out({"name": "y", "size": "err", "path": "/tmp/y"})
```

gives

```
Traceback (most recent call last):
  File "/home/kyle/src/python/pyout/pyout/field.py", line 210, in transform_fn
    return function(result)
  File "/home/kyle/src/python/pyout/venv-pyout/lib/python3.5/site-packages/humanize/filesize.py", line 25, in naturalsize
    bytes = float(value)
ValueError: could not convert string to float: 'err'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "transform_error.py", line 9, in <module>
    out({"name": "y", "size": "err", "path": "/tmp/y"})
  File "/home/kyle/src/python/pyout/pyout/tabular.py", line 486, in __call__
    self._check_and_write(row, style)
  File "/home/kyle/src/python/pyout/pyout/tabular.py", line 355, in _check_and_write
    self._set_widths(row, proc_group)
  File "/home/kyle/src/python/pyout/pyout/tabular.py", line 267, in _set_widths
    pre_keys=[proc_group], post_keys=[])
  File "/home/kyle/src/python/pyout/pyout/field.py", line 126, in __call__
    result = fn(value, result)
  File "/home/kyle/src/python/pyout/pyout/field.py", line 212, in transform_fn
    raise StyleFunctionError(function)
pyout.field.StyleFunctionError: Style transform <function naturalsize at 0x7f474cf8da60> raised an exception. See above.
```

So we report it as a StyleFunctionError but you can see the details of
the naturalsize ValueError above.  What is it that you want displayed?

Reply:
ah, I never thought of using "During handling of the above exception, another exception occurred"  "for profit" and just considered it of inferior coding style.  Not sure now -- seems to be informative etc.  So may be forget about this comment for now.

Reply:
> just considered it of inferior coding style

ha, it may be :)  For schema StyleError in element.py, I take care
cleaning up the previous exception and incorporating its message in
the StyleError:

https://github.com/pyout/pyout/blob/059e9a40c3579885991c0b3e4e1a597108b33869/pyout/elements.py#L160-L165


I guess I consider this different because the function that is raising
the exception is user-provided, so I feel like I shouldn't massage it
into something else.  But I might just be rationalizing the lazy
approach I took :)